snakemake/logging.py

```diff
@@ -14,6 +14,8 @@
 import tempfile
 from functools import partial
 import inspect
+import requests
```

Comment:
You need to move the requests import into the logger functions that use it. Otherwise, Python pickling will require it (even for unrelated cases) and it is not guaranteed to be there in every execution context (e.g. in a container).

Reply:
Thanks a lot. Indeed this was the issue.